warp10/src/main/java/io/warp10/script/functions/GOLDWRAP.java

```diff
@@ -0,0 +1,171 @@
+package io.warp10.script.functions;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.thrift.TDeserializer;
+import org.apache.thrift.TException;
+import org.apache.thrift.TSerializer;
+import org.apache.thrift.protocol.TCompactProtocol;
+
+import com.google.common.base.Charsets;
+
+import io.warp10.continuum.gts.GTSDecoder;
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GTSHelper;
+import io.warp10.continuum.gts.GTSWrapperHelper;
+import io.warp10.continuum.gts.GeoTimeSerie;
+import io.warp10.continuum.gts.GeoTimeSerie.TYPE;
+import io.warp10.continuum.store.thrift.data.GTSWrapper;
+import io.warp10.crypto.OrderPreservingBase64;
+import io.warp10.script.ElementOrListStackFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.ElementOrListStackFunction.ElementStackFunction;
+
+public class GOLDWRAP extends ElementOrListStackFunction {
+  
+  private final ElementStackFunction function;
+  
+  
+  public GOLDWRAP(String name) {
+    super(name);
+    function = generateFunctionOnce();
+  }
+
+  private ElementStackFunction generateFunctionOnce() {
+    return new ElementStackFunction() {
+      @Override
+      public Object applyOnElement(Object element) throws WarpScriptException {
+        GTSEncoder encoder = null;
+        
+        try {
+          if (element instanceof GeoTimeSerie) {
+            encoder = new GTSEncoder(0L);
+            encoder.encode((GeoTimeSerie) element);
```

Comment:
In that case, 'fullsort' and 'encodeOptimized' before doing the raw wrap could be faster.
However this very easy to read as is, not sure the modified version would be as easy to understand.

Reply:
Added a fast path.